### PR TITLE
Use injected DateTimeProvider in IBDState

### PR DESCRIPTION
**Pull request by codingupastorm**
The `InitialBlockDownloadState` shouldn't be using a different DateTimeProvider to that which is injected.

Remember to update either this or #3795 when either is merged.

### CODE REVIEW

**Comment by fassadlr** [comment]
@quantumagi can you update this PR to use the changes #3795 ?

File: src/Stratis.Bitcoin/Base/InitialBlockDownloadState.cs [hunk 1]
```diff
@@ -39,13 +39,13 @@ public class InitialBlockDownloadState : IInitialBlockDownloadState
...
         /// <param name="checkpoints">Provider of block header hash checkpoints.</param>
         /// <param name="loggerFactory">Provides us with a logger.</param>
-        public InitialBlockDownloadState(IChainState chainState, Network network, ConsensusSettings consensusSettings, ICheckpoints checkpoints, ILoggerFactory loggerFactory)
+        public InitialBlockDownloadState(IChainState chainState, Network network, ConsensusSettings consensusSettings, ICheckpoints checkpoints, ILoggerFactory loggerFactory, IDateTimeProvider dateTimeProvider)
         {
             this.network = network;
             this.consensusSettings = consensusSettings;
             this.chainState = chainState;
             this.checkpoints = checkpoints;
-            this.dateTimeProvider = DateTimeProvider.Default;
```
Comment (dangershony) [inline src/Stratis.Bitcoin/Base/InitialBlockDownloadState.cs]: What? how did that happen?

**Comment by codingupastorm** [comment]
I've merged to account for #3795 changes!

**Comment by codingupastorm** [comment]
Beware merge conflict with #3793

File: src/Stratis.Bitcoin/Consensus/ConsensusManager.cs [hunk 2]
```diff
@@ -1380,7 +1385,7 @@ private void AddComponentStats(StringBuilder log)
                 // gives us easier access to these values if issues are reported in the field.
 
                 // Use the default time provider - same as in InitialBlockDownloadState.IsInitialBlockDownload.
-                long currentTime = DateTimeProvider.Default.GetTime();
+                long currentTime =this.dateTimeProvider.GetTime();
```
Comment (quantumagi) [inline src/Stratis.Bitcoin/Consensus/ConsensusManager.cs]: Looks good. Just add " " after "=".

Comment (fassadlr) [inline src/Stratis.Bitcoin/Consensus/ConsensusManager.cs]:
```suggestion
                long currentTime = this.dateTimeProvider.GetTime();
```

**Comment by fassadlr** [comment]
@codingupastorm can you fix conflicts?

**Comment by fassadlr** [comment]
@codingupastorm as a final check can you perhaps sync TestNet and MainNet to 100k blocks and confirm here?

**Comment by codingupastorm** [comment]
Sure thing @fassadlr let me do this tomorrow

**Pull request merged**